tsl/src/fdw/relinfo.c

```diff
@@ -81,10 +81,15 @@ apply_fdw_and_server_options(TsFdwRelInfo *fpinfo)
 TsFdwRelInfo *
 fdw_relinfo_get(RelOptInfo *rel)
 {
-	TimescaleDBPrivate *rel_private = rel->fdw_private;
+	TimescaleDBPrivate *rel_private;
+
+	if (!rel->fdw_private)
+		rel->fdw_private = palloc0(sizeof(TimescaleDBPrivate));
```

Comment:
This will always populate the `rel->fdw_private` field. 

I wonder if this will have any unwanted side effects. So the question is; are there any issues with creating a zeroed out `fdw_private` on other code paths?

Reply:
we always create it like that so all the code should be able to handle it, also the Assert for fdw_private being NULL wasnt hit in my tests but it felt better to add this too since the assert below was hit and those two are similar and we can't say for sure if there isn't a query where fdw_private will be NULL